src/test-e2e/cloud-foundry/organizations-spaces-e2e.spec.ts

```diff
@@ -0,0 +1,187 @@
+import { CloudFoundryPage } from '../cloud-foundry/cloud-foundry.po';
+import { e2e } from '../e2e';
+import { EndpointsPage } from '../endpoints/endpoints.po';
+import { ConsoleUserType } from '../helpers/e2e-helpers';
+import { SideNavMenuItem, SideNavigation } from '../po/side-nav.po';
+import { StepperComponent } from '../po/stepper.po';
+import { ListComponent } from '../po/list.po';
+import { ConfirmDialogComponent } from '../po/confirm-dialog';
+import { MetaCard } from '../po/meta-card.po';
+import { CFHelpers } from '../helpers/cf-helpers';
+import { browser } from 'protractor';
+import { mergeEntity } from '../../frontend/app/store/helpers/reducer.helper';
+
+describe('CF - Manage Organizations and Spaces', () => {
+
+  const testOrgName = e2e.helper.getCustomerOrgSpaceLabel(null, 'org');
+  const testSpaceName = e2e.helper.getCustomerOrgSpaceLabel(null, 'space');
+  let endpointGuid;
+
+  let cloudFoundry: CloudFoundryPage;
+
+  let cfHelper: CFHelpers;
+
+  beforeAll(() => {
+    const setup = e2e.setup(ConsoleUserType.admin)
+      .clearAllEndpoints()
+      .registerDefaultCloudFoundry()
+      // Connect the test non-admin user to all cnsis in params
+      .connectAllEndpoints(ConsoleUserType.user)
+      // Connect the test admin user to all cnsis in params (required to ensure correct permissions are set when
+      // creating orgs + spaces)
+      .connectAllEndpoints(ConsoleUserType.admin)
+      .getInfo(ConsoleUserType.admin);
+
+    cfHelper = new CFHelpers(setup);
+
+  });
+
+  beforeEach(() => {
+    const endpointName = e2e.secrets.getDefaultCFEndpoint().name;
+    endpointGuid = e2e.helper.getEndpointGuid(e2e.info, endpointName);
+    // Go to the org view for this CF
+    cloudFoundry = CloudFoundryPage.forEndpoint(endpointGuid);
+    cloudFoundry.navigateTo();
+    cloudFoundry.waitForPageOrChildPage();
+  });
+
+  afterAll(() => {
+    return cfHelper.deleteSpaceIfExisting(endpointGuid, testSpaceName).then(() =>
+      cfHelper.deleteOrgIfExisting(endpointGuid, testOrgName)
+    );
+  });
+
+  it('Should validate org name', () => {
+    cloudFoundry.subHeader.clickItem('Organizations');
+
+    // Count the number of organizations
+    const cardView = new ListComponent();
+    cardView.cards.waitUntilShown();
+    cardView.cards.getCards().then(cards => {
```

Comment:
I think these lines should be refactored out to function since they are being repeated in most tests.

```
    const cardView = new ListComponent();
   cardView.cards.waitUntilShown();
    cardView.cards.getCards()
```